### ISSUE TITLE
Unable to use bot on StackExchange.com domain

### ISSUE THREAD

**Issue by Zomis** [Vincentyification/JavaBot#33]
It seems like the login forms does not look the same on StackOverflow.com and StackExchange.com, making it currently not possible to use this bot on a chat.stackexchange.com

**Comment by Vincentyification** [comment]
Hmm, this seems related to the Stack Exchange Chat authentication process. We are using HTMLUnit, which acts as a browser unit on its own. While Sam (poke: @ArcticEcho) has told me about a special authentication process being used in the Stack Exchange Chat (non-meta and non-SO), I doubt this is relevant. Maybe it will be an easy fix.

**Comment by Vincentyification** [comment]
I also hope the login form interfaces doesn't change in later builds, otherwise automatic searching for the button may be preferable.

**Comment by ArcticEcho** [comment]
... I was wondering when you guys would finally run into this problem. Well, to login into chat.SE it's quite simple really; just `POST` the user's `authToken` + `nonce` to `http://chat.stackexchange.com/users/login/global `. 

`nonce` can be fetched from `http://chat.stackexchange.com/users/login/global/request`, which returns JSON in the format of: `{"token":"xxxxxxxxxxxxxxxxxx","nonce":"xxxxxxxxxxxxxxxxxx"}` (the `token` here is somehow used (I'm still looking into this part) to finally get the user's `authToken` - all my attempts so far have proven unsuccessful).

**Comment by ArcticEcho** [comment]
Right, I've *finally* be able to get the chat.SE auth sorted out...

 1. Login to SE.

 2. `GET http://stackexchange.com/users/chat-login`.

 3. Get the value from the `value` attribute of the `input` elements named `authToken` and `nonce`.

 4. `POST` the above data (`authToken=blah-blah-blah&nonce=blah-blah-blah`) to `http://chat.stackexchange.com/users/login/global` with the request's referrer and origin both set to `http://chat.stackexchange.com`.

And that's it, you're in. Btw, the above `POST` request will return some JSON if the login was successful, example:

    {"Message":"Welcome \u003ca href=\"/users/117611/sam\" title=\"Sam\"\u003eSam\u003c/a\u003e, you\u0027ve been logged in."}

**Comment by Vogel612** [comment]
Ignore that one... that was meant for #34 ...
